src/frontends/tensorflow_common/src/op/select.cpp

```diff
@@ -54,27 +78,42 @@ OutputVector translate_select_op(const NodeContext& node) {
     // 1. Either the same shape (in which case the select is elementwise), or
     // 2. condition must be Rank 1 and match over the first dimension, or
     // 3. condition is scalar
-    default_op_checks(node, 3, {"Select"});
+    default_op_checks(node, 3, {"Select"}, true);
     auto condition = node.get_input(0);
     auto x = node.get_input(1);
     auto y = node.get_input(2);
-
-    // compute number of dimensions to unsqueeze the condition
-    auto cond_rank = compute_subgraph_scalar_rank(condition, element::i32);
+    auto complex_type_mark_x = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    auto complex_type_mark_y = as_type_ptr<ComplexTypeMark>(y.get_node_shared_ptr());
     auto x_rank = compute_subgraph_scalar_rank(x, element::i32);
-    auto num_new_axes = make_shared<v1::Subtract>(x_rank, cond_rank);
 
-    // generate a new shape for the condition
-    auto const_one = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
-    auto new_subshape = make_shared<v3::Broadcast>(const_one, num_new_axes);
-    auto cond_shape = make_shared<v3::ShapeOf>(condition, element::i32);
-    // use extra dimensions in the begin to avoid concatenation of empty tensors that is not supported by Concat
```

Comment:
please revert changes for non-complex type and have only changes for `if` branch. Now it is not convinient to review if we have regression for non-complex type